mailpoet/assets/js/src/automation/listing/index.tsx

```diff
@@ -47,14 +48,62 @@ const tableHeaders = [
   { key: 'more' },
 ] as const;
 
-export function AutomationListing({ workflows, loading }: Props): JSX.Element {
+export function AutomationListing({
+  workflows,
+  loading,
+  addNotice,
+}: Props): JSX.Element {
   const history = useHistory();
   const location = useLocation();
   const pageSearch = useMemo(
     () => new URLSearchParams(location.search),
     [location],
   );
 
+  const [allWorkflows, setAllWorkflows] = useState(workflows);
+
+  const updateWorkflow = useCallback(
+    (workflow: Workflow) => {
+      const newWorkflows = allWorkflows.map((existingWorkflow) => {
+        if (existingWorkflow.id === workflow.id) {
+          return workflow;
+        }
+
+        return existingWorkflow;
+      });
+      setAllWorkflows(newWorkflows);
+    },
+    [allWorkflows],
+  );
+
+  const removeWorkflow = useCallback(
+    (workflow: Workflow) => {
+      const newWorkflows = allWorkflows.filter(
+        (existingWorkflow) => existingWorkflow.id !== workflow.id,
+      );
+      setAllWorkflows(newWorkflows);
+    },
+    [allWorkflows],
+  );
+
+  const addWorkflow = useCallback(
+    (newWorkflow: Workflow) => {
+      const newWorkflows = [newWorkflow, ...allWorkflows];
+      setAllWorkflows(newWorkflows);
+    },
+    [allWorkflows],
+  );
+
+  const listingActions = useMemo<ListingActions>(
```

Comment:
This feels kind of silly, like I'm recreating a worse version of redux. Maybe we should just be using redux/wp data?

I played around with the idea of just having a "refreshWorkflows" action that would re-fetch all the workflows from the server, but it ended up feeling clunky. That being said, this way feels somewhat brittle.

Reply:
Yeah, I'll let you decide if you want to replace it with `@wordpress/data` now or if we should do it in another ticket. Now we don't have to hurry that much for the MVP deadline.